### PR TITLE
fix for promises to fix issue with msedge driver

### PR DESCRIPTION
**Pull request by senpl**
should fix https://github.com/webdriverio/selenium-standalone/issues/620#issuecomment-1048808131
or at least fix build on linux

### CODE REVIEW

File: lib/install-utils.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const mkdirp = require('mkdirp');
 const path = require('path');
 const yauzl = require('yauzl');
 const fs = require('fs');
+const fsPromises = require('fs').promises;
```
Comment (christian-bromann) [inline lib/install-utils.js:8]:
```suggestion
const fsPromises = require('fs/promises');
```

File: lib/install-utils.js [hunk 2]
```diff
@@ -66,20 +67,21 @@ const chmod = (where) =>
...
-    })
-  );
+  const requireChmod = await gotAccess(where)
 
   if (requireChmod) {
     await chmod(where);
   }
 }
 
+async function gotAccess(where) {
```
Comment (christian-bromann) [inline lib/install-utils.js:77]: This function will always return `undefined`, let's make sure we return something 😉

**Review by christian-bromann — CHANGES_REQUESTED** [review]

**Comment by christian-bromann** [comment]
@senpl can you check there seem to be a conflict in `lib/install-utils.js`

File: lib/install-utils.js (hunk 2 above)
Comment (senpl) [reply lib/install-utils.js:77]: I added returned value. it works on my machine.

**Comment by christian-bromann** [comment]
@senpl why do you think this fixes the issue we have in Linux? The current PR rewrites the same behavior just moves it out into a function.

**Comment by senpl** [comment]
> @senpl why do you think this fixes the issue we have in Linux? The current PR rewrites the same behavior just moves it out into a function.

When I submitted pull request it was not done in async way, and error from permission was send before other chcecks. When awaits are used error with file not downloaded seems to not be present on my machine. Still issue not got test case so it's hard to tell if this is what fix an issue. Anyway I wanted to fix issue in times when access check was not done in async way.

**Comment by senpl** [comment]
> @senpl why do you think this fixes the issue we have in Linux? The current PR rewrites the same behavior just moves it out into a function.

Also I am the one who report it. So If I get underline issue adding awaits fix part that not waiting for extraction a file. I checked latest version and the issue is fixed, so probably this merge is not needed. Still it fix issue for older version for me. So I could close it, as it's not really crucial now.

**Comment by christian-bromann** [comment]
Closing then. Thanks for looking into this anyway.

**Pull request closed without merging**